public/app/core/reducers/location.ts

```diff
@@ -13,26 +14,28 @@ export const initialState: LocationState = {
   lastUpdated: 0,
 };
 
-export const locationReducer = reducerFactory<LocationState>(initialState)
-  .addMapper({
-    filter: updateLocation,
-    mapper: (state, action): LocationState => {
-      const { path, routeParams, replace } = action.payload;
-      let query = action.payload.query || state.query;
+export const updateLocation = createAction<LocationUpdate>('location/updateLocation');
 
-      if (action.payload.partial) {
-        query = _.defaults(query, state.query);
-        query = _.omitBy(query, _.isNull);
-      }
+export const locationReducer = (state: LocationState = initialState, action: Action<unknown>) => {
```

Comment:
This state is mutated by Angular so converted it to a "regular" reducer

Reply:
why does this matter? Thought immer was only part of the reducer, and if angular modify state after this how come that has an impact?

Reply:
So ImmerJs has this `autoFreeze`option that freezes objects (https://github.com/reduxjs/redux-toolkit/issues/242) so state returned from `createSlice` or `createReducer` will be frozen so you can't change props on it. When I converted this reducer and the explore reducer with `createSlice` I got errors when running Grafana that Angular/Graph was trying to change read only props in state. Which is a nice feature if you want an immutable state, but in our case we kind of need a mutable state. In the explore case we could have a selector that copies the `graphResult`into a new object, but I didn't want to extend this PR further.

Reply:
> Can we have a comment here with the explanation?

Yes, I'll add a comment